### ISSUE TITLE
Access is allowed from event dispatch thread only

### ISSUE THREAD

**Issue by valis** [JetBrains/intellij-arend#369]
Should be an easy fix.

```
Access is allowed from event dispatch thread only

com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false
Current thread: Thread[JobScheduler FJ pool 9/15,4,main] 2071118376
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 559531093
	at com.intellij.openapi.application.impl.ApplicationImpl.throwThreadAccessException(ApplicationImpl.java:1103)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1086)
	at com.intellij.openapi.editor.impl.ScrollingModelImpl.assertIsDispatchThread(ScrollingModelImpl.java:158)
	at com.intellij.openapi.editor.impl.ScrollingModelImpl.getVisibleArea(ScrollingModelImpl.java:100)
	at com.intellij.openapi.editor.impl.softwrap.mapping.SoftWrapApplianceManager.updateAvailableArea(SoftWrapApplianceManager.java:471)
	at com.intellij.openapi.editor.impl.EditorImpl.lambda$initComponent$11(EditorImpl.java:1177)
	at java.desktop/java.awt.Component.processHierarchyEvent(Component.java:6821)
	at java.desktop/java.awt.Component.processEvent(Component.java:6440)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5029)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861)
	at java.desktop/java.awt.Component.removeNotify(Component.java:7218)
	at java.desktop/java.awt.Container.removeNotify(Container.java:2847)
	at java.desktop/javax.swing.JComponent.removeNotify(JComponent.java:4809)
	at java.desktop/java.awt.Container.removeNotify(Container.java:2831)
	at java.desktop/javax.swing.JComponent.removeNotify(JComponent.java:4809)
	at java.desktop/java.awt.Container.removeAll(Container.java:1322)
	at org.arend.toolWindow.errors.ArendMessagesView.updatePanel(ArendMessagesView.kt:255)
	at org.arend.toolWindow.errors.ArendMessagesView.updateEditors(ArendMessagesView.kt:202)
	at org.arend.toolWindow.errors.ArendMessagesView.valueChanged(ArendMessagesView.kt:159)
	at java.desktop/javax.swing.JTree.fireValueChanged(JTree.java:2968)
	at java.desktop/javax.swing.JTree$TreeSelectionRedirector.valueChanged(JTree.java:3457)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.fireValueChanged(DefaultTreeSelectionModel.java:641)
	at com.intellij.ui.treeStructure.Tree$MySelectionModel.fireValueChanged(Tree.java:619)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.clearSelection(DefaultTreeSelectionModel.java:580)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.rebuild(VariableHeightLayoutCache.java:743)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.treeStructureChanged(VariableHeightLayoutCache.java:629)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Handler.treeStructureChanged(BasicTreeUI.java:4324)
	at java.desktop/javax.swing.tree.DefaultTreeModel.fireTreeStructureChanged(DefaultTreeModel.java:607)
	at java.desktop/javax.swing.tree.DefaultTreeModel.reload(DefaultTreeModel.java:311)
	at java.desktop/javax.swing.tree.DefaultTreeModel.reload(DefaultTreeModel.java:218)
	at org.arend.toolWindow.errors.ArendMessagesView.update(ArendMessagesView.kt:333)
	at org.arend.toolWindow.errors.ArendMessagesService.update(ArendMessagesService.kt:58)
	at org.arend.highlight.TypecheckerPass$Companion.updateErrors(TypecheckerPass.kt:49)
	at org.arend.highlight.TypecheckerPassFactory.createPass(TypecheckerPassFactory.kt:28)
	at org.arend.highlight.TypecheckerPassFactory.createPass(TypecheckerPassFactory.kt:13)
	at org.arend.highlight.BasePassFactory.createHighlightingPass(BasePassFactory.kt:39)
	at com.intellij.codeInsight.daemon.impl.TextEditorHighlightingPassRegistrarImpl.instantiatePasses(TextEditorHighlightingPassRegistrarImpl.java:171)
	at com.intellij.codeInsight.daemon.impl.TextEditorBackgroundHighlighter.getPasses(TextEditorBackgroundHighlighter.java:75)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.lambda$queuePassesCreation$14(DaemonCodeAnalyzerImpl.java:960)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:933)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:68)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.lambda$queuePassesCreation$15(DaemonCodeAnalyzerImpl.java:951)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.lambda$queuePassesCreation$16(DaemonCodeAnalyzerImpl.java:943)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:184)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by knisht** [comment]
I have a reproducer:
```
\class R {
  \func<caret> f : Nat
}
```
Then press backspace.

**Comment by marat-rkh** [comment]
Fixed in [10194ff](https://github.com/JetBrains/intellij-arend/commit/10194ff0783baa5791506cdd4d3945d18d1f8f19).
